Geometry_Engine/Modify/SplitAtPoints.cs

```diff
@@ -212,111 +212,113 @@ public static List<PolyCurve> SplitAtPoints(this PolyCurve curve, List<Point> po
                 return new List<PolyCurve> { curve.DeepClone() };
 
             List<PolyCurve> result = new List<PolyCurve>();
-            List<ICurve> tmpResult = new List<ICurve>();
-            List<Point> subPoints = new List<Point>();
-            List<Point> onCurvePoints = new List<Point>();
 
-            foreach (Point p in points)
+            List<ICurve> subParts = curve.ISubParts().ToList(); //Subparts of PolyCurve
+            if (subParts.Count == 1)    //Single SubPart
             {
-                if (p.IsOnCurve(curve, tolerance))
-                    onCurvePoints.Add(p);
+                foreach (ICurve splitSegment in SplitAtPoints(subParts[0] as dynamic, points, tolerance))   //Simply call the appropriate split at method
+                {
+                    result.Add(new PolyCurve { Curves = new List<ICurve> { splitSegment } });   //And wrap split segments into polycurves
+                }
+                return result;  
             }
 
-            onCurvePoints = onCurvePoints.CullDuplicates(tolerance);
-
-            if (onCurvePoints.Count == 0)
-                return new List<PolyCurve> { curve.DeepClone() };
-
-            onCurvePoints = onCurvePoints.SortAlongCurve(curve);
-
-            foreach (ICurve crv in curve.SubParts())
+            List<Point> nonDuplicatePoints = points.CullDuplicates(tolerance);
```

Comment:
Shouldn't this happen at the beginning? Why is it call for multiple segments but not for one?

Reply:
Mainly because those methods call `CullDuplicates` themselves anyway. Hence calling it here would be a wastefull call that would be of no real benefit.

Reply:
Right, makes sense. However, if it is a question of just one call then I would still opt for generalising the method for n >= 1 segments instead of the individual treatment of n == 1. But happy for you to make a final decision, you def have a better feeling as the originator 👍

Reply:
Honestly, had it like that at first. (no special case). Main reason I added this was due to issues with closed single segmented subparts (read circle) where the strategy for prev stopped working as it lept around and attached to itself. 

Can be handled in other ways ofc, but this felt like the simplest way of handling it..

Reply:
fixed now. No special case for single curve